SpatialGDK/Source/SpatialGDK/Private/Utils/CheckoutRadiusConstraintUtils.cpp

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "Utils/CheckoutRadiusConstraintUtils.h"
+
+#include "UObject/UObjectIterator.h"
+
+#include "SpatialGDKSettings.h"
+
+DEFINE_LOG_CATEGORY(LogCheckoutRadiusConstraintUtils);
+
+namespace SpatialGDK
+{
+
+QueryConstraint CheckoutRadiusConstraintUtils::GetDefaultCheckoutRadiusConstraint()
+{
+	const float MaxDistanceSquared = GetDefault<USpatialGDKSettings>()->MaxNetCullDistanceSquared;
+
+	// Use AActor's ClientInterestDistance for the default radius (all actors in that radius will be checked out)
+	const AActor* DefaultActor = Cast<AActor>(AActor::StaticClass()->GetDefaultObject());
+
+	float DefaultDistanceSquared = DefaultActor->NetCullDistanceSquared;
+
+	if (MaxDistanceSquared != 0.f && DefaultDistanceSquared > MaxDistanceSquared) {
+		UE_LOG(LogCheckoutRadiusConstraintUtils, Warning, TEXT("Default NetCullDistanceSquared is too large, clamping from %f to %f"),
+			DefaultDistanceSquared, MaxDistanceSquared);
+
+		DefaultDistanceSquared = MaxDistanceSquared;
+	}
```

Comment:
Nit: could this be moved to a function `GetNetCullDistanceSquaredClampedForActor()`, since this code appears twice in this file?

Reply:
it could be the same, I only didn't do it because it would make building the log message slightly more awkward for the same amount of info. happy to change it

Reply:
Ah yes, the log is slightly different. It's fine to keep it as it is then.